### PR TITLE
Allow fission in fixed source simulations

### PR DESCRIPTION
**Pull request by paulromano**
Now that we have the capability to put secondary particles in a bank, in principle there is no reason we can't do that for a fixed source simulation. This pull request enables fission neutrons to be generated and stored in the secondary bank in a fixed source simulation. Of course, if you have a critical or supercritical system, your secondary bank will blow up and the run will end. Since the secondary bank has room for 100 particles right now, this means that subcritical multiplication problems will work up to k-eff=0.99 since M=1/(1-k)=1/(1-0.99)=100.

### CODE REVIEW

File: src/physics.F90 [hunk 1]
```diff
@@ -88,9 +88,14 @@ subroutine sample_reaction(p)
...
 
-    if (nuc % fissionable .and. run_mode == MODE_EIGENVALUE) then
+    if (nuc % fissionable) then
       call sample_fission(i_nuclide, i_reaction)
-      call create_fission_sites(p, i_nuclide, i_reaction)
+      if (run_mode == MODE_EIGENVALUE) then
+        call create_fission_sites(p, i_nuclide, i_reaction, fission_bank, n_bank)
+      elseif (run_mode == MODE_FIXEDSOURCE) then
+        call create_fission_sites(p, i_nuclide, i_reaction, &
+             p%secondary_bank, p%n_secondary)
```
Comment (wbinventor) [inline src/physics.F90]: space around the `%`

File: src/physics.F90 [hunk 2]
```diff
@@ -1153,17 +1168,16 @@ subroutine create_fission_sites(p, i_nuclide, i_reaction)
...
-      fission_bank(i) % uvw(3) = sqrt(ONE - mu*mu) * sin(phi)
+      bank_array(i) % uvw(1) = mu
+      bank_array(i) % uvw(2) = sqrt(ONE - mu*mu) * cos(phi)
+      bank_array(i) % uvw(3) = sqrt(ONE - mu*mu) * sin(phi)
 
       ! Sample secondary energy distribution for fission reaction and set energy
       ! in fission bank
-      fission_bank(i) % E = sample_fission_energy(nuc, nuc%reactions(&
-           i_reaction), p)
+      bank_array(i) % E = sample_fission_energy(nuc, nuc%reactions(i_reaction), p)
```
Comment (wbinventor) [inline src/physics.F90]: space between the `%`

**Comment by wbinventor** [comment]
This is a straightforward PR to me. I'm fine merging this in unless you want someone more knowledgeable about the source bank to do the honors.

**Comment by paulromano** [comment]
@wbinventor I agree it is straightforward enough.

**Pull request merged**